### DISCUSSION TITLE
How to share a single server connection across different clients using mitmproxy?

### DISCUSSION THREAD

**Discussion by Ryan-gsq** [mitmproxy/mitmproxy#7095] [Q&A]
Hello mitmproxy Community,

I am currently working on a project where I need to allow multiple clients to share a single server connection using mitmproxy. The goal is to intercept and modify HTTP requests and responses for multiple clients, but I want all these clients to communicate through the same server connection to optimize resource usage and maintain consistency.

### Scenario:

- Multiple clients send requests to various services.
- One mitmproxy instance intercepts all these requests.

- All requests should be sent through a single persistent server connection.

### Questions:

1. Is it possible to configure mitmproxy to allow multiple clients to share a single server connection?
2. If possible, what are the steps or configurations needed to implement this setup?

3. Are there any potential issues or limitations I should be aware of when using this approach?

### Attempts:

- I have tried looking into the documentation but couldn't find specific information on this scenario.
- I experimented with different settings in mitmproxy but didn't achieve the desired outcome.

Any guidance or suggestions would be greatly appreciated. Thank you for your help!

**Comment by mhils** [comment]
> Is it possible to configure mitmproxy to allow multiple clients to share a single server connection?

No, not really.

**Comment by rosydawn6** [comment]
Is your requirement that all clients use the same and a single "mode"?  If not I have found that the when starting one mitmproxy server with multiple listeners via the --mode command the incremental increase in memory/cpu for each mode is not too high when compared with the initial startup cost (memory, cpu) of mitmproxy itself.

**Comment by Ryan-gsq** [comment]
thanks
